packages/fxa-auth-server/lib/payments/stripe.ts

```diff
@@ -446,47 +423,47 @@ class StripeHelper {
     return customer;
   }
 
+  static customerCacheKey = (args: any[]) => `customer-${args[0]}|${args[1]}`;
+
+  @Cacheable({ cacheKey: StripeHelper.customerCacheKey })
+  async cachedCustomer(
+    uid: string,
+    email: string
+  ): Promise<Stripe.Customer | void> {
+    return this.fetchCustomer(uid, email, [
+      'data.sources',
+      'data.subscriptions',
+      'data.invoice_settings.default_payment_method',
+    ]);
+  }
+
   /**
    * Fetch a customer for the record from Stripe based on user ID & email.
    *
    * Uses Redis caching if configured.
    */
-  async customer(
-    uid: string,
-    email: string,
+  async customer({
+    uid,
+    email,
     forceRefresh = false,
-    cacheOnly = false
-  ): Promise<Stripe.Customer | undefined> {
-    const cacheKey = this.customerCacheKey(uid, email);
-    let result = undefined;
-
-    if (!forceRefresh && this.redis) {
-      try {
-        const json = await this.redis.get(cacheKey);
-        if (json) {
-          result = JSON.parse(json);
-        }
-      } catch (err) {
-        this.log.error(`stripeHelper.customer.redis.get.failed`, { err });
-      }
+    cacheOnly = false,
+  }: {
+    uid: string;
+    email: string;
+    forceRefresh?: boolean;
+    cacheOnly?: boolean;
```

Comment:
I'll admit its not the most beautiful thing in TypeScript to specify the type and deconstruct it like this in one go. I switched to an object here because it was becoming increasingly confusing in `subscriptions.js` and elsewhere to see 2 boolean args at the end, and not be entirely sure what it was going to do. Switching to an object here made the caller's much more readable and clear.